api/app/jobs/model_data.md

```diff
@@ -0,0 +1,122 @@
+<!-- omit in toc -->
+# Fetching & Interpreting Weather Model Data
+
+<!-- omit in toc -->
+## Table of Contents
+
+- [Introduction](#introduction)
+- [GRIB Files](#grib-files)
+- [Data Sources](#data-sources)
+  - [Environment Canada](#environment-canada)
+    - [Relevant Data Links](#relevant-data-links)
+      - [GDPS](#gdps)
+      - [RDPS](#rdps)
+      - [HRDPS](#hrdps)
+    - [Relevant Weather Variables](#relevant-weather-variables)
+  - [NOAA](#noaa)
+      - [Model Run Cycles](#model-run-cycles)
+- [Downloading Data](#downloading-data)
+- [Linear Interpolation](#linear-interpolation)
+  - [From grids to fire weather stations](#from-grids-to-fire-weather-stations)
+  - [To 12:00 PST (13:00 PDT)](#to-1200-pst-1300-pdt)
+- [Cronjobs](#cronjobs)
+
+
+## Introduction
+
+Some of the core functionalities of Morecast (versions 1 and 2) require that we regularly fetch weather model data from various sources. These weather models are computer-generated weather forecasts (although for the sake of clarity, throughout this project we refer to them as "weather models", whereas "forecasts" strictly apply to the weather predictions made by human weather forecasters). Third-party organizations run these weather models on their own hardware and publish the output publicly. Morecast then regularly consumes this weather model data and performs additional analysis on it to suit our specific needs.
+
+## GRIB Files
+
+All weather model produce outputs in GRIB file format. [Refer here for a description of what a GRIB file is](https://weather.gc.ca/grib/what_is_GRIB_e.html).
+
+A GRIB file contains one or more raster bands, in which weather data is presented in a grid format. Each raster band corresponds to one weather variable. (For example, temperature data points will be stored in a different raster band to wind speed data points.) In order to coalesce the points on the data grid with geographic locations (such as lat/long coordinates), a geographic coordinate transformation must be applied. This coordinate transformation is performed in our backend, using metadata supplied by the source of the GRIB file.
```

Comment:
This is a great overview, thank you. Are the coordinates of the data grid just (metric, value)? e.g. (temp, 21.2)?

For the geographic transform, is that based on a raster band in the grid file whose coordinates are (lat, long)?

Reply:
I'm not sure what the data grid overall looks like. AFAIK it's not possible to read from the datagrid directly. Instead we use gdal to retrieve a specific raster band from the data grid, where the id of the raster band corresponds to the weather variable we want. That then returns a 2D array of values for the weather variable. So ultimately we do something like `datagrid.GetRasterBand(581)` where 581 is the ID of the `TMP_TGL_2m` weather variable, and that returns something like

```
[[21.2, 21.4, 21.6, ...],
[20.9. 21.5, 23.4, ...],
...
]
```

I'll add a little bit more to the doc to explain this.